File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.swift
import Foundation

func solution(_ A: [Int], _ B: [Int]) -> Int {
    var ans = 0

    var pseudoA = A
    var pseudoB = B
    
    pseudoA.sort()
    pseudoB.sort(by: >)
    
    for i in 0..<A.count {
        ans += pseudoA[i] * pseudoB[i]
    }

    return ans
}

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> A, vector<int> B)
{
    const int n = A.size();
    int answer = 0;
    sort(A.begin(), A.end());
    sort(B.begin(), B.end(), greater<int>());
    
    for(int i = 0; i < n; i++) {
        answer += A[i] * B[i];
    }

    return answer;
}